src/test/scala/io/flow/apidoc-json-validation/JsonValidatorSpec.scala

```diff
@@ -78,6 +79,53 @@ class JsonValidatorSpec extends FunSpec with Matchers {
     )
   }
 
+  it("converts 'number' into a string where possible when number is a double") {
+    val form = Json.obj(
+      "url" -> 123.45,
+      "events" -> Seq("*")
+    )
+    validator.validate("webhook_form", form) should equal(
+      Right(
+        Json.obj(
+          "url" -> "123.45",
+          "events" -> Seq("*")
+        )
+      )
+    )
+  }
+
+  it("validates a double") {
+    validator.validate("double", Json.parse("123.45")).right.get.as[Double] should equal(123.45)
+    validator.validate("double", Json.parse("123")).right.get.as[Double] should equal(123)
+    validator.validate("double", JsString(" ")) should equal(Left(List("Type 'double' must be a valid double")))
+  }
+
+  it("validates a decimal") {
+    validator.validate("decimal", Json.parse("123.45")).right.get.as[BigDecimal] should equal(123.45)
+    validator.validate("decimal", Json.parse("123")).right.get.as[BigDecimal] should equal(123)
+    validator.validate("decimal", JsString(" ")) should equal(Left(List("Type 'decimal' must be a valid decimal")))
+  }
+
+  it("validates a UUID") {
+    val uuid = java.util.UUID.randomUUID
+    validator.validate("uuid", JsString(uuid.toString)).right.get.as[java.util.UUID] should equal(uuid)
+    validator.validate("uuid", JsString(" ")) should equal(Left(List("Type 'uuid' must be a valid UUID")))
+  }
+
+  it("validates an ISO 8601 date (yyyy-MM-dd)") {
+    validator.validate("date-iso8601", JsString("2017-02-24")).right.get.as[String] should equal("2017-02-24")
+    validator.validate("date-iso8601", JsString("invalid")) should equal(Left(List("Type 'date-iso8601' must be a valid ISO 8601 date")))
+    // Tests that the format must be yyyy-MM-dd
+    validator.validate("date-iso8601", JsString((new DateTime(2017, 2, 24, 0, 0, 0)).toString)) should equal(Left(List("Type 'date-iso8601' must be a valid ISO 8601 date")))
+  }
+
+  it("validates an ISO 8601 datetime") {
+    val dt = (new DateTime(2017, 2, 24, 0, 0, 0)).toString
+    validator.validate("date-time-iso8601", JsString("2017-02-24")).right.get.as[String] should equal(dt)
```

Comment:
It looks like the parser assumes the time is 00:00:00, which makes sense to me since we're upcasting a `yearMonthDay` (for lack of a better term) to a `dateTime`. I think it's okay, especially if the generated clients use this method to parse `DateTime` objects:
```
scala> ISODateTimeFormat.dateTimeParser.parseDateTime(new DateTime(2017, 2, 24, 0, 0, 0).toString)
res2: org.joda.time.DateTime = 2017-02-24T00:00:00.000-05:00

scala> ISODateTimeFormat.dateTimeParser.parseDateTime("2017-02-24")
res3: org.joda.time.DateTime = 2017-02-24T00:00:00.000-05:00

scala> ISODateTimeFormat.dateTimeParser.parseDateTime("2017-2-24")
res4: org.joda.time.DateTime = 2017-02-24T00:00:00.000-05:00
```